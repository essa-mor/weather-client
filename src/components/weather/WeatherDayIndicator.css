.weather_day_indicator{
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: space-around;
	height: 90px;
	width: 73px;
	cursor: pointer;
}

.weather_day_indicator_selected{
	background-color: #fcfcfc;
    border: 1px solid #e9e9e9;
    border-radius: 1px;
}

.weather_day{
	margin-top: 5px;
	font-family: arial, sans-serif;
	font-size: 13px;
	color: #bababa ;
}

.weather_temperatures{
	margin-bottom: 5px;
}
.weather_max_temperature{
	margin-right: 5px;
	font-family: arial, sans-serif;
	font-size: 13px;
	color: #878787 ;
}

.weather_min_temperature{
	font-family: arial, sans-serif;
	font-size: 13px;
	color: #bababa ;
}