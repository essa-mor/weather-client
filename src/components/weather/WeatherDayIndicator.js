import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import dateFormat from 'dateformat';

import './WeatherDayIndicator.css';

function handleOnClick(callBack, dt){
	 if(callBack != null){
		 callBack(dt);
	 }
}

function WeatherDayIndicator({ onClick, isSelected, dt, day, img, maxTemperature, minTemperature }) {
	const cls = classNames('weather_day_indicator', { 'weather_day_indicator_selected': isSelected });
	return (<div className={cls} onClick={handleOnClick.bind(this, onClick, dt)}>
		<span className="weather_day">{dateFormat(new Date(day), 'ddd')}</span>
		<img src={img} alt="" />
		<div className="weather_temperatures">
			<span className="weather_max_temperature">{Math.round(maxTemperature)}°</span>
			<span className="weather_min_temperature">{Math.round(minTemperature)}°</span>
		</div>
	</div>);
}

WeatherDayIndicator.propTypes = {
	onClick: PropTypes.func,
	isSelected: PropTypes.bool,
	dt: PropTypes.number.isRequired,
	day: PropTypes.string.isRequired,
	img: PropTypes.string.isRequired,
	maxTemperature: PropTypes.number.isRequired,
	minTemperature: PropTypes.number.isRequired
};

export default WeatherDayIndicator;