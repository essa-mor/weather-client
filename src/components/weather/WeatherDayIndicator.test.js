import React from 'react';
import renderer from 'react-test-renderer';

import WeatherDayIndicator from './WeatherDayIndicator';

describe('WeatherDayIndicator test', () => {
	it('renders correctly', () => {
		const tree = renderer.create(<WeatherDayIndicator
			day="Wen"
			img="https://ssl.gstatic.com/onebox/weather/48/partly_cloudy.png"
			maxTemperature={25}
			minTemperature={20} />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('renders correctly - selected', () => {
		const tree = renderer.create(<WeatherDayIndicator
			isSelected={true}
			day="Wen"
			img="https://ssl.gstatic.com/onebox/weather/48/partly_cloudy.png"
			maxTemperature={25}
			minTemperature={20} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});