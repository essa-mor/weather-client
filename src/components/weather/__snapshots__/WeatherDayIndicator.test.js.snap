// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherDayIndicator test renders correctly - selected 1`] = `
<div
  className="weather_day_indicator weather_day_indicator_selected"
  onClick={[Function]}
>
  <span
    className="weather_day"
  >
    Fri
  </span>
  <img
    alt=""
    src="https://ssl.gstatic.com/onebox/weather/48/partly_cloudy.png"
  />
  <div
    className="weather_temperatures"
  >
    <span
      className="weather_max_temperature"
    >
      25
      째
    </span>
    <span
      className="weather_min_temperature"
    >
      20
      째
    </span>
  </div>
</div>
`;

exports[`WeatherDayIndicator test renders correctly 1`] = `
<div
  className="weather_day_indicator"
  onClick={[Function]}
>
  <span
    className="weather_day"
  >
    Thu
  </span>
  <img
    alt=""
    src="https://ssl.gstatic.com/onebox/weather/48/partly_cloudy.png"
  />
  <div
    className="weather_temperatures"
  >
    <span
      className="weather_max_temperature"
    >
      25
      째
    </span>
    <span
      className="weather_min_temperature"
    >
      20
      째
    </span>
  </div>
</div>
`;
